conftest.py

```diff
@@ -26,9 +26,12 @@
 
 
 def pytest_sessionstart():
-    """ Download sct_testing_data prior to test collection. """
-    logger.info("Downloading sct test data")
-    downloader.main(['-d', 'sct_testing_data', '-o', sct_test_path()])
+    """Download sct_testing_data prior to test collection."""
+    # pytest_sessionstart runs once before any test session starts, then again for each
+    # worker session. We only want it to run once, so check to see if we're in a worker session.
+    if "PYTEST_XDIST_WORKER" not in os.environ.keys():
+        logger.info("Downloading sct test data")
+        downloader.main(['-d', 'sct_testing_data', '-o', sct_test_path()])
```

Comment:
With this new solution, `pytest_cases` (and the other test changes) aren't strictly necessary, since `sct_testing_data/` will be available before test collection...

But, I'd love to hear feedback on this from @kousu.

Reply:
I'm glad you discovered that this works. But despite its shortcomings, I think sticking closer to what pytest wants us to do will be easier. Making sure test data is in place is the first use-case fixtures get taught for in my experience. I answered some of this above too: https://github.com/neuropoly/spinalcordtoolbox/pull/3152#discussion_r554485231